tests/test_dimension.py

```diff
@@ -830,6 +831,76 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_condition_lowering(self):
+        """
+        Test the lowering of a ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+
+        # Define Function
+        g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+        g.data[:] = 1
+        x, y = grid.dimensions
+        ci = ConditionalDimension(name='ci', parent=y, condition=Le(g, 1))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f, g)).apply()
```

Comment:
you could also `assert` against the expected structure of the lowered condition, ie (pseudocode)

```
assert condition.lhs.indices == ...
```

Reply:
WIll do.

Reply:
OK, thx

Reply:
Now, the lowering is not happening upon creation. I added another test for lowering+index-shifting of the relationals. Not yet for CDs although I can do as well.